app/controllers/pages_controller.rb

```diff
@@ -1,11 +1,35 @@
 class PagesController < ApplicationController
   include PagesHelper
+  include CitiesHelper
   respond_to :html, :json
 
 #TODO add here where users are active this year
   def home
+
+    # get region id
+    region_id = params[:region_id] ? params[:region_id] : ""
+
+    # get country_id
+    country_id = params[:country_id] ? params[:country_id] : ""
+
+    @cities = City
+    if !country_id.present? && !region_id.present?
+      @cities = @cities.all
+    else
+      @cities = filter_cities(@cities, country_id, region_id)
+      @cities = @cities.sort_alphabetical_by{ |c| c[:name] }
+    end
+    # get city ids
+    city_ids = nil
+    if region_id.present?
+      city_ids = @cities.map {|c| c[:id] }
+    end
+
   	@hosts = Host.includes(:city, :user, :country, :invites).where(host_conditions_hash)
-  	@hosts = @hosts.select { |h| 
+    @hosts = @hosts.where(city_id: city_ids) if city_ids != nil
+  	@hosts = @hosts.paginate(:page => params[:page] || 1, :per_page => 10)
+    @total_items = @hosts.count
+    @hosts = @hosts.select { |h|
```

Comment:
remember that `select` is slow and if possible, only call conditional when needed